docs/data_classes.rst

```diff
@@ -0,0 +1,301 @@
+
+.. code-block:: python
+
+  >>> linewidth_wcs_dict_reversed = {
+  ...    'CTYPE2': 'HPLT-TAN', 'CUNIT2': 'deg', 'CDELT2': 0.5, 'CRPIX2': 2, 'CRVAL2': 0.5, 'NAXIS2': 20,
+  ...    'CTYPE1': 'HPLN-TAN', 'CUNIT1': 'deg', 'CDELT1': 0.4, 'CRPIX1': 2, 'CRVAL1': 1, 'NAXIS1': 10}
+  >>> linewidth_wcs_reversed = astropy.wcs.WCS(linewidth_wcs_dict_reversed)
+  >>> linewidth_cube_reversed = NDCube(linewidth_data.transpose(), linewidth_wcs_reversed)
+
+We can still define an `~ndcube.NDCollection` with aligned axes by supplying a tuple of tuples, giving the aligned axes of each ND object separately.  In this case, the 1st axis of the ``observations`` cube is aligned with the 2nd axis of the ``linewidths`` cube and vice versa.
+
+.. code-block:: python
+
+   >>> my_collection_reversed = NDCollection(
+   ...    [("observations", my_cube), ("linewidths", linewidth_cube_reversed)],
+   ...    aligned_axes=((0, 1), (1, 0)))
+   >>> my_collection_reversed.aligned_axes
+   {'observations': (0, 1), 'linewidths': (1, 0)}
+
+Because aligned axes must have the same lengths, we can get the lengths of the aligned axes by using the ``aligned_dimensions`` property.
+
+.. code-block:: python
+
+  >>> my_collection.aligned_dimensions
+  <Quantity [3., 4.] pix>
+
+Note that this only tells us the lengths of the aligned axes.  To see the lengths of the non-aligned axes, e.g. the spectral axis of the ``observations`` object, you must inspect that ND object individually.
+
+We can also see the physical properties to which the aligned axes correspond by using the `~ndcube.NDCollection.aligned_axis_physical_types` property.
+
+.. code-block:: python
+
+  >>> my_collection.aligned_axis_physical_types  # doctest: +SKIP
+  [('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat'), ('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat')]
+
+This returns the a `list` of `tuple` giving the physical types that correspond to each aligned axis.  For each aligned axis, only physical types are associated with all the cubes in the collection are returned.  Note that there is no there is no requirement that all aligned axes must represent the same physical types.  They just have to be the same length.  Therefore, is it possible that this property returns no physical types.
+
+The real power behing `~ndcube.NDCollection.aligned_axes` is that is enables all objects within the `~ndcube.NDCollection` to be sliced along the aligned axes simultaneously form the `~ndcube.NDCollection` level.  This allows users to quickly and accurately crop their entire data set to a region of interest, thereby speeding up their analysis workflow.  See the :ref:`collection_slicing` to see this in action.
```

Comment:
provide an example of this?

Reply:
Examples are shown in the slicing section of the docs.

Reply:
I did miss that, but I do wonder if it's worth a very terse example in line?